Minio/MinioClient.cs

```diff
@@ -142,6 +145,42 @@ private async Task<string> GetRegion(string bucketName)
             return (region == string.Empty) ? "us-east-1" : region;
         }
 
+        /// <summary>
+        /// Constructs a RestRequest. For AWS, this function has the side-effect of overriding the baseUrl
+        /// in the RestClient with region specific host path or virtual style path.
+        /// </summary>
+        /// <param name="args">The  object specifically MakeBucketArgs class, args with populated values. From Input</param>
+        /// <param name="method">The Method needed to call with this request object</param>
+        /// <returns>A RestRequest</returns>
+        internal RestRequest CreateRequest(MakeBucketArgs args, Method method)
```

Comment:
why is this `MakeBucketArgs`?

Reply:
if MakeBucket needs to be treated specially, do so within MakeBucketAsync - this generic method is not required.